mousestyles/path_diversity/path_features_advanced.py

```diff
@@ -0,0 +1,287 @@
+
+    if len(path_obj) <= 1:
+        raise ValueError("path_obj must contain at least 2 rows")
+
+    if not isinstance(center, dict):
+        raise TypeError("center must be a dictionary")
+
+    if set(center.keys()) != {'x', 'y'}:
+        raise ValueError("the keys of center must be 'x', and 'y'")
+
+    indices = path_obj.index[:-1]
+    vectors = [ path_obj.loc[i,'x':'y'] - [center['x'],center['y']] for i in indices]
+    # center_angles = [oath_features.angle_between(v1,v2) for v1,v2 in zip(vectors[1:], vectors[:len(vectors)])]
+    center_angles = [angle_between(list(v1),list(v2)) for v1,v2 in zip(vectors[1:], vectors[:-1])]
+    radius = [np.linalg.norm(v) for v in vectors] 
+    return({'radius': radius, 'center_angles': center_angles})
+
+def compute_area_covered(r_and_theta):
+    r"""
+    Returns the area spanned by the path.
+
+    For 2 adjacent points in a path and its center, the area of the triangle
+    constructed by such 3 points can be calculated by the 2 radius and the center
+    angle: let r1 and r2 be 2 adjacent radius, and theta be the angle between such 2 vectors.
+    The area of the triangle is then calculated by
+    r1 * r2 * sin(theta) / 2.
+
+    This function sums up such areas of the triangles over all pairs of 2 adjacent points
+    in the path.
+
+    Parameters
+    ----------
+    r_and_theta : dictionary
+        its key must be radius and center_angles. 
+        Expecting the output of compute_area_radius_and_center_angle.
+
+    Returns
+    -------
+    area : numpy float
+        area computed by radius and center angles in the path.
+
+    Examples
+    --------
+    >>> movement = data.load_movement(1,2,1)
+    >>> sep = path_index(movement, 1, 1)
+    >>> path = movement[sep[2][0]:sep[2][1]+1]
+    >>> edges = find_edge_points(path)
+    >>> cent = find_center(edges)
+    >>> r_and_ca = compute_radius_and_center_angle(path, cent)
+    >>> area = compute_area_covered(r_and_ca)
+    """
+
+    if not isinstance(r_and_theta, dict):
+        raise TypeError("r_and_theta must be a dictionary")
+
+    if set(r_and_theta.keys()) != {'radius', 'center_angles'}:
+        raise ValueError("the keys of r_and_theta must be 'radius' and 'center_angles'")
+
+    if not isinstance(r_and_theta['radius'], list):
+        raise TypeError("r_and_theta['radius'] must be a list")
+
+    if not isinstance(r_and_theta['center_angles'], list):
+        raise TypeError("r_and_theta['center_angles'] must be a list")
+
+    if len(r_and_theta['radius']) != len(r_and_theta['center_angles']) + 1:
+        raise ValueError("length of r_and_theta['radius'] must be \
+            the length of r_and_theta['center_angles'] plus 1")
+
+    zipped = zip(r_and_theta['radius'][1:], r_and_theta['radius'][:-1], \
+                 r_and_theta['center_angles'])
+    areas = [ v1 * v2 * np.sin(theta) / 2 for v1,v2,theta in zipped]
+    return(sum(areas))
+
+def compute_start_end_distance(path_obj):
```

Comment:
This function does not calculate the total length of the path nor the distances between adjacent 2 points, but does calculate the length of the straight line which directly connects the starting and ending points in a path. In other words, we ignore the intermediate points and only focus on the 2 points, the 1st and the last. I think this can't be done by `get_dist_and_speed` made by @jcpalumbo3 and @sicihuang.